repobuilder/job_test.go

```diff
@@ -73,3 +73,60 @@ func TestProcessPackages(t *testing.T) {
 	assert.NoError(t, j.processPackages(ctx))
 	assert.Len(t, j.PackagePaths, 6)
 }
+
+func TestGetPackageLocation(t *testing.T) {
+	for _, test := range []struct {
+		name             string
+		version          string
+		expectedLocation string
+	}{
+		{
+			name:             "LegacyReleaseCandidate",
+			version:          "4.2.5-rc1",
+			expectedLocation: "testing",
+		},
+		{
+			name:             "LegacyDevelopmentBuild",
+			version:          "4.2.5-pre-",
+			expectedLocation: "development",
+		},
+		{
+			name:             "LegacyDevelopmentSeries",
+			version:          "4.2.5",
+			expectedLocation: "4.2",
+		},
+		{
+			name:             "LegacyProductionSeries",
+			version:          "4.1.5",
+			expectedLocation: "4.1",
+		},
+		{
+			name:             "NewReleaseCandidate",
+			version:          "5.3.5-rc1",
+			expectedLocation: "testing",
+		},
+		{
+			name:             "NewDevelopmentBuild",
+			version:          "5.0.5-alpha1",
+			expectedLocation: "development",
+		},
+		{
+			name:             "NewQuarterly",
```

Comment:
If it's quarterly, there won't be a "major.minor.5", since the yearly cycle rolls over after major.minor.3".

Reply:
I thought minor was the one that rolled over at 3, and the patch number could be anything.

Reply:
True, nvm the new terminology is so confusing.

Reply:
Yeah, doesn't minor represent the quarter?